frontend/src/pages/ProjectListPage.jsx

```diff
@@ -10,27 +11,86 @@ import ProjectName from "@components/project/ProjectName"
 import ProjectEdit from "@components/project/edit/ProjectEdit"
 import SkeletonProjectList from "@components/project/skeletons/SkeletonProjectList"
 
-import { getProjectList } from "@api/projects.api"
+import { getProjectList, patchProject } from "@api/projects.api"
 
 import { ifMobile } from "@utils/useScreenType"
 
+import queryClient from "@queries/queryClient"
+
 import FeatherIcon from "feather-icons-react"
+import { HTML5Backend } from "react-dnd-html5-backend"
+import {
+    DndProvider,
+    MouseTransition,
+    MultiBackend,
+    TouchTransition,
+} from "react-dnd-multi-backend"
+import { TouchBackend } from "react-dnd-touch-backend"
 import { useTranslation } from "react-i18next"
 
 const ProjectListPage = () => {
     const { t } = useTranslation(null, { keyPrefix: "project_list" })
 
-    const {
-        isPending,
-        isError,
-        data: projects,
-        refetch,
-    } = useQuery({
+    const [projects, setProjects] = useState([])
+    const [isCreateOpen, setIsCreateOpen] = useState(false)
+
+    const { isPending, isError, data, refetch } = useQuery({
         queryKey: ["projects"],
         queryFn: () => getProjectList(),
     })
 
-    const [isCreateOpen, setIsCreateOpen] = useState(false)
+    useEffect(() => {
+        if (!data) return
+        setProjects(data)
+    }, [data])
+
+    const patchMutation = useMutation({
+        mutationFn: ({ id, order }) => {
+            return patchProject(id, { order })
+        },
+        onSuccess: () => {
+            queryClient.invalidateQueries({
+                queryKey: ["projects"],
+            })
+            refetch()
+        },
+    })
+
+    const moveProject = useCallback((dragIndex, hoverIndex) => {
+        setProjects((prevProjects) => {
+            const updatedProjects = [...prevProjects]
+            const [moved] = updatedProjects.splice(dragIndex, 1)
+            updatedProjects.splice(hoverIndex, 0, moved)
+            return updatedProjects
+        })
+    }, [])
+
+    const dropProject = useCallback(() => {
+        const changedProjects = projects
+            .map((project, index) => ({ id: project.id, order: index }))
+            .filter((project, index) => data[index]?.id !== project.id)
+
+        changedProjects.forEach(({ id, order }) => {
+            patchMutation.mutate({ id, order })
+        })
```

Comment:
`changedProjects` 요소의 필드에는 `id`와 `order`뿐이니 오브젝트 분해가 없어도 괜찮을 것 같습니다.

```suggestion
        changedProjects.forEach((element) => {
            patchMutation.mutate(element)
        })
```